File: python/problema7.py
'''
7. Utilize a função map que receba como parâmetros uma sequência numérica e uma
 função lambda. A função lambda deve retornar par ou ı́mpar para cada número.
'''

paridade = map(lambda e: "par" if e%2 == 0 else "impar", [2,3,4,5,6,7])
print(list(paridade))

File: HaskelAtividades/HaskellLambda/problema7.hs
{-
Utilize a fun ̧c ̃aomapque recebe como parˆametros uma lista num ́erica e
uma fun ̧c ̃ao lambda.  A fun ̧c ̃aolambda deve retornar par ou  ́ımpar
para cada n ́umero, ou seja, uma lista de booleans.
-}

main = do
  print ( (map (\x -> mod x 2==0) [0,1,2,3,4,5]) )
